### DISCUSSION TITLE
When should users not commit `.yarn/cache`

### DISCUSSION THREAD

**Discussion by benatshippabo** [yarnpkg/berry#4155] [Q&A] [answered]
Hi, after some reading my understanding is that the cache in `.yarn/cache` is highly optimized. But it seems that it is only recommended to be committed into git when using `nodeLinker: pnp`. My question is why is it also not recommended to commit it when using other `nodeLinkers` like `node-modules`? It seems like this would speed up CI without much downside.

**Comment by arcanis** [comment, marked as answer]
You can. It also protects you against the npm registry being temporarily down, so personally I'd see it as a good practice in enterprise settings where reliability is important, but that's up to you.

**Reply by benatshippabo** [reply]
Oh I see! Thanks for the swift reply @arcanis ❤️
